crates/circuit/src/circuit_data.rs

```diff
@@ -995,6 +996,16 @@ impl CircuitData {
         self.param_table.clear();
     }
 
+    pub fn count_ops(&mut self) -> IndexMap<String, i32> {
+        let mut ops_count: IndexMap<String, i32> = IndexMap::new();
```

Comment:
One thing you should do here is set the hasher to `ahash::RandomState` with:

```suggestion
        let mut ops_count: IndexMap<String, i32, ::ahash::RandomState> = IndexMap::default();
```

it performs better than the default hash algorithm.